apps/prairielearn/src/sync/course-db.ts

```diff
@@ -1334,24 +1150,29 @@ async function validateAssessment(
     });
 
     const validateViewAndSubmitRolePermissions = (
-      canView: string[] | null | undefined,
-      canSubmit: string[] | null | undefined,
+      canView: Set<string> | null | undefined,
+      canSubmit: Set<string> | null | undefined,
```

Comment:
👀 it looks like these are now sets? I see usage of `z.set(z.string())` in some places. How does this work? I tried executing `z.set(z.string()).parse(["foo", "foo"])` and I got an error saying `Expected set, received array`.

Reply:
Fixed in [0756a14](https://github.com/PrairieLearn/PrairieLearn/pull/11024/commits/0756a147abffc99fa3065ee0de2a2f1e52b1056d).

Reply:
Let's continue discussion here: https://github.com/PrairieLearn/PrairieLearn/pull/11024#discussion_r1907917205